### PR TITLE
[imp] fieldservice stock

### PR DESCRIPTION
**Pull request by smangukiya**
ref: https://github.com/OCA/field-service/issues/65

@max3903 : Please review it. Thanks

### CODE REVIEW

File: fieldservice_stock/models/fsm_equipment.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    current_stock_location_id = fields.Many2one(
+        'stock.location', string='Current Inventory Location',
+        compute='_get_current_stock_loc_id')
+
+    @api.multi
+    def _get_current_stock_loc_id(self):
+        for equipment in self:
+            if equipment.lot_id and equipment.lot_id.product_id:
+                equipment.current_stock_location_id = \
+                    equipment.lot_id.product_id.property_stock_inventory \
```
Comment (max3903) [inline fieldservice_stock/models/fsm_equipment.py]: @smangukiya `current_location_location_id` should be set to the current location of the serial number: when the serial number is delivered to the customer location, the current location should change to `Partner Location / Customers`.

File: fieldservice_stock/models/fsm_order.py [hunk 2]
```diff
@@ -35,15 +35,27 @@ def _default_warehouse_id(self):
...
+                 picking.picking_type_id.code == 'outgoing'])
+            order.return_count = len(
+                [picking for picking in order.picking_ids if
+                 picking.picking_type_id.code == 'incoming'])
 
     def action_confirm(self):
         if self.location_id and self.line_ids and self.warehouse_id:
             self.line_ids._confirm_picking()
+            if self.return_ids:
+                self.return_ids._confirm_picking()
```
Comment (max3903) [inline fieldservice_stock/models/fsm_order.py]: @smangukiya It doesn't work if I have a return line (inbound) and no line (outbound). I keep getting the error message below. Confirming should confirm the draft lines and draft returns if there is any.

**Review by max3903 — CHANGES_REQUESTED** [review]

File: fieldservice_stock/models/fsm_order.py (hunk 2 above)
Comment (max3903) [reply fieldservice_stock/models/fsm_order.py]: @smangukiya I fixed the logic but we still need to filter the lines and returns with draft ones.

File: fieldservice_stock/models/fsm_equipment.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+from odoo import api, fields, models
+
+
+class FSMEquipment(models.Model):
+    _inherit = 'fsm.equipment'
+    _description = "FSM Equipment Inherits"
+
+    lot_id = fields.Many2one(
+        'stock.production.lot', string='Lot/Serial #')
```
Comment (max3903) [inline fieldservice_stock/models/fsm_equipment.py]: @smangukiya No need for "Lot/". We need a `product_id` field to filter the lot_id. Both fields should be required.

**Comment by smangukiya** [comment]
@max3903 : Source has been improved as per your comments:
1. No need for "Lot/". We need a product_id field to filter the lot_id. Both fields should be required
2. I fixed the logic but we still need to filter the lines and returns with draft ones.
3. current_location_location_id should be set to the current location of the serial number: when the serial number is delivered to the customer location, the current location should change to Partner Location / Customers.

**Review by max3903 — APPROVED** [review]

**Pull request merged**